client/my-sites/plans/current-plan/main.jsx

```diff
@@ -39,12 +43,47 @@ const PlanDetailsComponent = React.createClass( {
 		sitePlans: React.PropTypes.object.isRequired,
 		fetchPlans: React.PropTypes.func.isRequired
 	},
+
 	componentWillUpdate: function( props ) {
 		this.props.fetchPlans( props );
 	},
+
 	componentDidMount: function() {
 		this.props.fetchPlans();
 	},
+
+	getPurchaseInfo: function() {
+		const plan = this.props.currentPlan;
+
+		if ( ! plan ) {
+			return null;
+		}
+
+		const hasAutoRenew = plan.auto_renew;
+		const classes = classNames( 'current-plan__purchase-info', {
+			'is-expiring': plan.userFacingExpiryMoment < this.moment().add( 30, 'days' )
```

Comment:
Perhaps we can use a selector instead for the`is-expiring` value. It will make it easier to remove the moment dates from the plans redux tree in the future.